lightgbm_ranker.py

```diff
@@ -0,0 +1,180 @@
+        self.params = params
+        self.model_path = model_path
+        self.query_features = query_features
+        self.match_features = match_features
+        self.categorical_query_features = categorical_query_features
+        self.categorical_match_features = categorical_match_features
+        self.label = label
+        if self.model_path and os.path.exists(self.model_path):
+            self.booster = lightgbm.Booster(model_file=self.model_path)
+            model_num_features = self.booster.num_feature()
+            expected_num_features = len(self.query_features + self.match_features)
+            if categorical_query_features:
+                expected_num_features += len(categorical_query_features)
+            if categorical_match_features:
+                expected_num_features += len(categorical_match_features)
+            if model_num_features != expected_num_features:
+                raise ValueError(
+                    f'The number of features expected by the LightGBM model {model_num_features} is different'
+                    f'than the ones provided in input {expected_num_features}'
+                )
+        else:
+            self.booster = None
+
+    def _get_features_dataset(
+        self, docs: DocumentArray, task: str = 'predict'
+    ) -> 'lightgbm.Dataset':
+        q_features, m_features, group, labels = [], [], [], []
+        query_feature_names = self.query_features
+        if self.categorical_query_features:
+            query_feature_names += self.categorical_query_features
+        match_feature_names = self.match_features
+        if self.categorical_match_features:
+            match_feature_names += self.categorical_match_features
+        for doc in docs:
+            query_feature = []
+            match_feature = []
+            query_values = [doc.tags.get(feature) for feature in query_feature_names]
+            for match in doc.matches:
+                match_values = [
+                    match.tags.get(feature) for feature in match_feature_names
+                ]
+                labels.append(match.tags.get(self.label))
+                match_feature.append(match_values)
+                query_feature.append(query_values)
+            group.append(len(doc.matches))
+            q_features.append(query_feature)
+            m_features.append(match_feature)
+
+        query_dataset = lightgbm.Dataset(
+            data=np.vstack(q_features),
+            group=group,
+            feature_name=query_feature_names,
+            categorical_feature=self.categorical_query_features,
+            free_raw_data=False,
+        )
+
+        match_dataset = lightgbm.Dataset(
+            data=np.vstack(m_features),
+            group=group,
+            label=labels,
+            feature_name=match_feature_names,
+            categorical_feature=self.categorical_match_features,
+            free_raw_data=False,
+        )
+        if task == 'predict':
```

Comment:
what is this task?

Reply:
@JoanFM I replaced [this](https://github.com/jina-ai/jina-hub/blob/28bc886f6568aa95b61b0fe5fe47c473e8d633e2/rankers/LightGBMRanker/__init__.py#L25) with `task` here

Reply:
and why is this a `task`? where did u get the inspiration from?